source/code/providers/Container_HostInventory_Class_Provider.cpp

```diff
@@ -0,0 +1,212 @@
+            {
+                //swarm
+                hostInventoryInstance.NodeRole_value("Master");  
+            }
+            else if(computerName.find("agent") != string::npos)
+            {
+                //swarm
+                hostInventoryInstance.NodeRole_value("Agent");  
+            }
+            else
+            {
+                hostInventoryInstance.NodeRole_value("Not Orchestrated");  
+            }   
+
+            //Orchestrator type is obtained from the node name. example of node names swarm-manager-vmss_0 for swarm, k8s-master-71E8D996-0 for kubernetes
+            if(computerName.find("swarmm") != string::npos)
+            {
+                //swarm
+                hostInventoryInstance.OrchestratorType_value("Swarm Mode");  
+            }
+            else if(computerName.find("swarm") != string::npos)
+            {
+                //swarm
+                hostInventoryInstance.OrchestratorType_value("Swarm");  
+            }
+            else if(computerName.find("k8") != string::npos)
+            {
+                //swarm
+                hostInventoryInstance.OrchestratorType_value("Kubernetes");  
+            }
+            else if(computerName.find("dcos") != string::npos)
+            {
+                //swarm
+                hostInventoryInstance.OrchestratorType_value("DC/OS");  
+            }
+            else
+            {
+                hostInventoryInstance.OrchestratorType_value("None");  
+            }   
+                        
+            // Clean up object
+            cJSON_Delete(response[0]);
+        }
+        else
+        {
+            syslog(LOG_WARNING, "API call in Container_HostInventory::InspectHost to get host info returned null");
+        }
+        return hostInventoryInstance;
+    }
+};
+
+Container_HostInventory_Class_Provider::Container_HostInventory_Class_Provider(
+    Module* module) :
+    m_Module(module)
+{
+}
+
+Container_HostInventory_Class_Provider::~Container_HostInventory_Class_Provider()
+{
+}
+
+void Container_HostInventory_Class_Provider::Load(
+        Context& context)
+{
+    context.Post(MI_RESULT_OK);
+}
+
+void Container_HostInventory_Class_Provider::Unload(
+        Context& context)
+{
+    context.Post(MI_RESULT_OK);
+}
+
+void Container_HostInventory_Class_Provider::EnumerateInstances(
+    Context& context,
+    const String& nameSpace,
+    const PropertySet& propertySet,
+    bool keysOnly,
+    const MI_Filter* filter)
+{
+    printf("Container_HostInventory_Class_Provider ei \n");
+    Container_HostInventory_Class queryResult = ContainerHostInventoryQuery::InspectHost();
+    context.Post(queryResult);
+    context.Post(MI_RESULT_OK);
+}
+
+void Container_HostInventory_Class_Provider::GetInstance(
+    Context& context,
+    const String& nameSpace,
+    const Container_HostInventory_Class& instanceName,
+    const PropertySet& propertySet)
+{
+    context.Post(MI_RESULT_NOT_SUPPORTED);
```

Comment:
You don't want to support this for efficiency? I did note that you do have a key in the new class, so this can be supported (if it would ever be used). In general, we support these from [SCXcore](https://github.com/Microsoft/SCXcore).